spec/models/theme_spec.rb

```diff
@@ -68,14 +68,23 @@
   end
 
   it 'can correctly find parent themes' do
-    grandchild = Theme.create!(name: 'grandchild', user_id: user.id)
     child = Theme.create!(name: 'child', user_id: user.id)
     theme = Theme.create!(name: 'theme', user_id: user.id)
 
     theme.add_child_theme!(child)
-    child.add_child_theme!(grandchild)
 
-    expect(grandchild.dependant_themes.length).to eq(2)
+    expect(child.dependant_themes.length).to eq(1)
+  end
+
+  it "doesn't allow multi-level theme components" do
+    grandchild = Theme.create!(name: 'grandchild', user_id: user.id)
+    child = Theme.create!(name: 'child', user_id: user.id)
+    theme = Theme.create!(name: 'theme', user_id: user.id)
+
+    theme.add_child_theme!(child)
+    expect do
+      child.add_child_theme!(grandchild)
+    end.to raise_error(Discourse::InvalidParameters)
```

Comment:
We should validate for the error message here as well. Note that `raise_error` takes the expected error message as the second parameter.